src/main/java/frc/robot/subsystems/Drive.java

```diff
@@ -2,33 +2,69 @@
 
 import java.util.function.DoubleSupplier;
 
+import com.chopshop166.chopshoplib.maps.DifferentialDriveMap;
 import com.chopshop166.chopshoplib.outputs.SendableSpeedController;
 
 import edu.wpi.first.wpilibj.drive.DifferentialDrive;
 import edu.wpi.first.wpilibj2.command.CommandBase;
 import edu.wpi.first.wpilibj2.command.RunCommand;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
-import frc.robot.maps.RobotMap;
 
+/**
+ * 1) What does it do? Makes motors turn a certain amount depending on how much
+ * an axis is triggered
+ * 
+ * 2) What modes does it have? Not at the moment
+ * 
+ * 3) What interactions does it have with other subsystems? Does not have any
+ * interactions with other systems other than the space it shares on the robot
+ * 
+ * 4) How is it triggered -> OI? No triggers runs by default
+ * 
+ * 5) Does it store any state? Right and Left motor groups.
+ * 
+ * 6) Sensors? Encoders, Gyro
+ */
 public class Drive extends SubsystemBase {
 
     private final SendableSpeedController rightMotorGroup;
     private final SendableSpeedController leftMotorGroup;
     private final DifferentialDrive driveTrain;
 
-    public Drive(RobotMap.DriveMap map) {
+    /**
+     * Defines the left and right motor(s) and then puts them into a differential
+     * drive
+     * 
+     * @param map represents the drive map
+     */
+    public Drive(DifferentialDriveMap map) {
         super();
-        rightMotorGroup = map.right();
-        leftMotorGroup = map.left();
+        rightMotorGroup = map.getRight();
+        leftMotorGroup = map.getLeft();
         driveTrain = new DifferentialDrive(leftMotorGroup, rightMotorGroup);
     }
 
+    /**
+     * Maps the drive axises
+     * 
+     * @param forward defines the forward direction
+     * @param turn    defines the direction to turn
+     * @return returns a run command so drive will stay running as long as drive is
+     *         being called
+     */
     public CommandBase drive(DoubleSupplier forward, DoubleSupplier turn) {
         return new RunCommand(() -> {
             double yAxis = forward.getAsDouble();
             double xAxis = turn.getAsDouble();
-
             driveTrain.arcadeDrive(yAxis, xAxis);
         }, this);
     }
+
+    public CommandBase driveFlip(DoubleSupplier forward, DoubleSupplier turn) {
```

Comment:
This should have the same comment structure as the `drive` command. 

It should also be bound to a button with `toggleWhenActive`.